1.train_models/utils/GreedySubsetSum.py

```diff
@@ -0,0 +1,116 @@
+import pandas as pd
+from PlateTrainingSplits import PlateTrainingSplits
+
+
+class GreedySubsetSum:
+
+    def __init__(self):
+        pass
+
+    def update_test_wells(self, _welldf, _category_col, _well_col, _cell_count_col, _test_well_count):
+        """
+        Parameters
+        ----------
+        _welldf: Pandas Dataframe
+            Well data with unique wells represented as rows.
+
+        _category_col: List of Strings or String
+            The categories to represent equally in the cell population.
+
+        _well_col: String
+            Well column name.
+
+        _cell_count_col: String
+            The cell count column name.
+
+        _test_well_count: Integer
+            Number of wells to sample for testing.
+
+        Returns
+        -------
+        test_wells: List of Strings
+            Test well names.
+        """
+
+        # Check if all of the cell counts are integers.
+        if not pd.api.types.is_integer_dtype(_welldf[_cell_count_col]):
+            raise TypeError(f"{_cell_count_col} column does not only contain integers")
+
+        # Check if all of the integers are greater than zero
+        if (_welldf[_cell_count_col] <= 0).all():
+            raise ValueError(f"{_cell_count_col} column contains only positive integers")
+
+        plate_split = PlateTrainingSplits()
+
+        catdf = _welldf.groupby(_category_col)
+
+        # Determine the smalles reference category for sampling test wells
+        min_cat = plate_split.sum_train_val_category_cell_counts(
+            catdf,
+            _category_col,
+            _cell_count_col,
+            _test_well_count
+        )
```

Comment:
Could this result possibly not be a reference category but rather the smallest category?

Reply:
Unless reference in this case is a technical reference group not a biological reference group? If so, consider add a few words to the inline comments to make this more clear?

Reply:
> Could this result possibly not be a reference category but rather the smallest category?

It should be both.

> Unless reference in this case is a technical reference group not a biological reference group? If so, consider add a few words to the inline comments to make this more clear?

The original intention was distinguish between biological groups to create a technical adjustment. Group are only constrained to specific columns in this implementation, so it's possible to also define these as technical groups.